packages/vscode-extension/src/debug/taskTerminal/devTunnelTaskTerminal.ts

```diff
@@ -132,7 +132,7 @@ export class DevTunnelTaskTerminal extends BaseTunnelTaskTerminal {
     await this.outputStartMessage(devTunnelDisplayMessages);
     await this.outputStartDevTunnelStepMessage(devTunnelDisplayMessages);
     await this.resolveArgs(this.args);
-    // TODO: delete the last debug tunnel if it is not deleted
+    await this.deleteExistingTunnel();
```

Comment:
How long does delete operation usually take? If deletion is slow, should we do it asynchronously (without await)?

Reply:
> How long does delete operation usually take? If deletion is slow, should we do it asynchronously (without await)?
It is about 0.2s.

Once the maximum number of tunnels is reached, the existing tunnel should be deleted before they can create a new one.

Reply:
At present, Basis limits each user to a maximum of 10 tunnels, and in the future, the number of tunnels available for the free tier may be further reduced. To ensure that the previous tunnel is closed correctly and avoid reaching the maximum number of tunnels (may be similiar to ngrok, allow only 2 tunnel to be opened), I recommend deleting the tunnel before creating a new one, as long as a delay of 0.2 seconds is acceptable